src/rpc/net.cpp

```diff
@@ -538,6 +540,327 @@ static RPCHelpMan getaddednodeinfo()
+                               if (v.msg_count > 0) {
+                                   UniValue msg_stats(UniValue::VOBJ);
+                                   msg_stats.pushKV("message_count", v.msg_count.load());
+                                   msg_stats.pushKV("byte_count", v.byte_count.load());
+                                   result.pushKV(k, msg_stats);
+                               }
+                           }
+                       },
+                       [&](const std::map<std::string, MultiLevelStatsMap>& m) {
+                           for (auto& [k, v] : m) {
+                               UniValue v_univalue = v.to_univalue();
+                               if (v_univalue.size() > 0) {
+                                   result.pushKV(k, v_univalue);
+                               }
+                           }
+                       }},
+                   m_varmap);
+        return result;
+    }
+};
+
+/**
+ * Arrange the stats in `stats.m_data` according to the `dimension_types`. Sum all the
+ * data for dimensions that are not listed.
+ * @code
+ * sent                      recv                      (direction)
+ * ||___________             ||___________
+ * |            |            |            |
+ * tor          ipv4         tor          ipv4         (network)
+ * ||_____      ||_____      ||_____      ||_____
+ * |      |     |      |     |      |     |      |
+ * verack ping  verack ping  verack ping  verack ping  (message type)
+ * |      |     |      |     |      |     |      |
+ * a      b     c      d     e      f     g      h
+ * @endcode
+ *
+ * If the direction and network dimensions are requested, the stats for message type are
+ * summarized and the above would become:
+ * @code
+ * sent                      recv                      (direction)
+ * ||___________             ||___________
+ * |            |            |            |
+ * tor          ipv4         tor          ipv4         (network)
+ * |            |            |            |
+ * *            *            *            *
+ * |            |            |            |
+ * a+b          c+d          e+f          g+h          (message type)
+ * @endcode
+ *
+ * This method can also rearrange results by dimension, allowing the requester to specify
+ * the order. If the message type and network dimensions are requested, the response
+ * would be:
+ * @code
+ * verack                    ping                      (message type)
+ * ||___________             ||___________
+ * |            |            |            |
+ * tor          ipv4         tor          ipv4         (network)
+ * |            |            |            |
+ * *            *            *            *
+ * |            |            |            |
+ * a+e          c+g          b+f          d+h          (direction)
+ * @endcode
+ *
+ */
+UniValue Aggregate(const std::vector<DimensionType>& dimensions, const NetStats::MultiDimensionalStats& raw_stats)
```

Comment:
I realized that this can be done in a simpler way - if the stats are put in a map and `""` used for the aggregated dimensions. See commit b7c56affe9 `rpc: make it possible to aggregate the result in getnetmsgstats` from [`vasild/getnetmsgstats`](https://github.com/vasild/bitcoin/commits/getnetmsgstats).